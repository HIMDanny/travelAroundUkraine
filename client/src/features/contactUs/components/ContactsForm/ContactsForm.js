import React, { useState, memo } from 'react';
import { Button, styled, Typography } from '@mui/material';
import { TextField } from 'formik-mui';
import { Formik, Form, Field } from 'formik';
import * as yup from 'yup';
import { AlertModal } from '../../../../components';
import subscribeUser from '../../../../api/subscribeUser';

const TextFieldContactsPersonData = styled(Field)(({ theme }) => ({
  width: '270px',

  [theme.breakpoints.up('tablet')]: {
    width: '280px',
  },
  [theme.breakpoints.up('laptop')]: {
    width: '300px',
  },
}));

const ContactsForm = () => {
  const [isModalOpen, setIsModalOpen] = useState(false);
  const [modalTitle, setModalTitle] = useState('');
  const [modalText, setModalText] = useState('');
  const [modalSuccess, setModalSuccess] = useState('true');

  const closeModal = () => setIsModalOpen(false);

  const initialValues = {
    email: '',
  };

  const schema = yup.object().shape({
    email: yup.string('Enter your email').email('Enter a valid email').required('Email is required'),
  });

  const handleSubmit = async (values) => {
    const { isSuccess, error } = await subscribeUser(values);
    if (isSuccess) {
      setIsModalOpen(true);
      setModalTitle('Subscription Successful');
      setModalText('You can enjoy our Newsletter service now on your mail');
      setModalSuccess('true');
    } else {
      setIsModalOpen(true);
      setModalTitle('Subscription failed');
      setModalText(error.message);
      setModalSuccess('false');
    }
  };

  return (
    <Formik initialValues={initialValues} validationSchema={schema} onSubmit={handleSubmit}>
      {() => (
        <Form>
          <TextFieldContactsPersonData component={TextField} name="email" label="Email" margin="normal" />

          <Button margin="normal" sx={{ display: 'block', marginTop: '15px' }} type="submit">
            Subscribe
          </Button>

          <AlertModal
            open={isModalOpen}
            onClose={closeModal}
            onSubmit={closeModal}
            title={modalTitle}
            disableCancelButton
            submitButtonText="Close"
            success={modalSuccess}
          >
            <Typography>{modalText}</Typography>
          </AlertModal>
        </Form>
      )}
    </Formik>
  );
};

export default memo(ContactsForm);
