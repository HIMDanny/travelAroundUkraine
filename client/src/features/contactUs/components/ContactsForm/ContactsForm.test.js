/* eslint-disable react/prop-types */
import { render } from '@testing-library/react';
import ContactsForm from './ContactsForm';

jest.mock('@mui/material/TextField', () => ({ children }) => <field>{children}</field>);

describe('ContactsForm snapshot test', () => {
  test('should ContactsForm match snapshot', () => {
    const { asFragment } = render(<ContactsForm />);
    expect(asFragment()).toMatchSnapshot();
  });
});