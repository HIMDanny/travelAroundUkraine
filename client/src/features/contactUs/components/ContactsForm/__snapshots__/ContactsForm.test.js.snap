// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactsForm snapshot test should ContactsForm match snapshot 1`] = `
<DocumentFragment>
  <form
    action="#"
  >
    <field />
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1ey3g8u-MuiButtonBase-root-MuiButton-root"
      margin="normal"
      tabindex="0"
      type="submit"
    >
      Subscribe
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </form>
</DocumentFragment>
`;
